nexus/db-queries/src/db/datastore/saga.rs

```diff
@@ -207,16 +200,73 @@ impl DataStore {
 
         Ok(events)
     }
+
+    /// Updates all sagas that are currently assigned to any of the Nexus
+    /// instances in `nexus_zone_ids`, assigning them to `new_sec_id` instead.
+    ///
+    /// This change causes the Nexus instance `new_sec_id` to discover these
+    /// sagas and resume executing them the next time it performs saga recovery
+    /// (which is normally on startup and periodically).  Generally,
+    /// `new_sec_id` is the _current_ Nexus instance and the caller should
+    /// activate the saga recovery background task after calling this function
+    /// to immediately resume the newly-assigned sagas.
+    ///
+    /// **Warning:** This operation is only safe if the Nexus instances
+    /// `nexus_zone_ids` are not currently running.  If those Nexus instances
+    /// are still running, then two (or more) Nexus instances may wind up
+    /// running the same saga concurrently.  This would likely violate implicit
+    /// assumptions made by various saga actions, leading to hard-to-debug
+    /// errors and state corruption.
+    pub async fn sagas_reassign_sec(
+        &self,
+        opctx: &OpContext,
+        nexus_zone_ids: &[db::saga_types::SecId],
```

Comment:
Why call them `nexus_zone_ids` and not `current_sec_ids` or similar?

Reply:
Yeah, this was a little self-inconsistent.  I've changed this in 87e57183e and updated the comment so it's less flip-flopping between Nexus ids and SEC ids.